File: app/src/main/java/net/north101/android/ghplayertracker/data/PerkItem.java
package net.north101.android.ghplayertracker.data;

import org.json.JSONException;
import org.json.JSONObject;
import org.parceler.Parcel;
import org.parceler.ParcelConstructor;

@Parcel
public class PerkItem {
    public final PerkAction perkAction;
    public final String cardId;
    public final int repeat;

    @ParcelConstructor
    public PerkItem(PerkAction perkAction, String cardId, int repeat) {
        this.perkAction = perkAction;
        this.cardId = cardId;
        this.repeat = repeat;
    }

    public static PerkItem parse(JSONObject jsonObject) throws JSONException {
        PerkAction perkAction = PerkAction.parse(jsonObject.getString("action"));
        int repeat = jsonObject.optInt("repeat", 1);
        String cardId = jsonObject.getString("card_id");

        return new PerkItem(perkAction, cardId, repeat);
    }
}

File: app/src/main/java/net/north101/android/ghplayertracker/data/PerkItem.kt
package net.north101.android.ghplayertracker.data

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize
import org.json.JSONException
import org.json.JSONObject

@Parcelize
class PerkItem(
        val perkAction: PerkAction,
        val cardId: String,
        val repeat: Int
) : Parcelable {
    companion object {
        @Throws(JSONException::class)
        fun parse(jsonObject: JSONObject): PerkItem {
            val perkAction = PerkAction.parse(jsonObject.getString("action"))
            val repeat = jsonObject.optInt("repeat", 1)
            val cardId = jsonObject.getString("card_id")

            return PerkItem(perkAction, cardId, repeat)
        }
    }
}
